### ISSUE TITLE
error: ‘mono_aot_register_module’ undeclared

### ISSUE THREAD

**Issue by Ronaldho80** [mono/mono#11526]
## Steps to Reproduce

1. This part of my Makefile:
```
# all dependent assemblies                                                                                                         
 15 #                                                                                                                                  
 14 SOURCES=Zlib.cs GZipCompressor.cs SGML.cs Program.cs                                                                               
 13                                                                                                                                    
 12 DEPS=packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.Data.dll \                                              
 11      packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.RawFileReader.dll \                                     
 10      packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.MassPrecisionEstimator.dll \                            
  9      packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.Data.dll \                                              
  8      packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.BackgroundSubtraction.dll \                             
  7      packages/NDesk.Options.0.2.1/lib/NDesk.Options.dll \                                                                          
  6      packages/Ionic.Zip.1.9.1.8/lib/Ionic.Zip.dll                                                                                  
  5                                                                                                                                    
  4 # add '-r:' option to each assembly                                                                                                
  3 PACKAGES=$(addprefix -r:,$(DEPS))                                                                                                  
  2                                                                                                                                    
  1 bundle: compile                                                                                                                    
  0     mkbundle --nodeps --static -o RawConverter2 RawConvert2.exe -L /usr/lib/mono/4.5/ $(DEPS)                                      
```

### Current Behavior
Error making the bundle: 
```
herzog@Elfriede:~/Lipotype/RawConverter2 
$ make
mcs -o -debug- -r:packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.Data.dll -r:packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.RawFileReader.dll -r:packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.MassPrecisionEstimator.dll -r:packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.Data.dll -r:packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.BackgroundSubtraction.dll -r:packages/NDesk.Options.0.2.1/lib/NDesk.Options.dll -r:packages/Ionic.Zip.1.9.1.8/lib/Ionic.Zip.dll Zlib.cs GZipCompressor.cs SGML.cs Program.cs -out:RawConvert2.exe
warning CS8029: Compatibility: Use -out:FILE instead of --output FILE or -o FILE
mkbundle --nodeps --static -o RawConverter2 RawConvert2.exe -L /usr/lib/mono/4.5/ packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.Data.dll packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.RawFileReader.dll packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.MassPrecisionEstimator.dll packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.Data.dll packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.BackgroundSubtraction.dll packages/NDesk.Options.0.2.1/lib/NDesk.Options.dll packages/Ionic.Zip.1.9.1.8/lib/Ionic.Zip.dll
temp.c: In function ‘init_default_mono_api_struct’:
temp.c:134:33: warning: assignment from incompatible pointer type [-Wincompatible-pointer-types]
  mono_api.mono_jit_set_aot_mode = mono_jit_set_aot_mode;
                                 ^
temp.c:135:38: error: ‘mono_aot_register_module’ undeclared (first use in this function)
  mono_api.mono_aot_register_module = mono_aot_register_module;
                                      ^
temp.c:135:38: note: each undeclared identifier is reported only once for each function it appears in
temp.c: At top level:
temp.c:110:1: warning: ‘validate_api_struct’ defined but not used [-Wunused-function]
 validate_api_struct ()
 ^
temp.c:129:1: warning: ‘init_default_mono_api_struct’ defined but not used [-Wunused-function]
 init_default_mono_api_struct ()
 ^
ERROR: [Fail] 256
Makefile:17: recipe for target 'bundle' failed
make: *** [bundle] Error 1

```

### Expected Behavior
It compiles a bundle

## On which platforms did you notice this

[ ] macOS
[ x] Linux
[ ] Windows

**Version Used**:
```
herzog@Elfriede:~/Lipotype/RawConverter2 
$ mono --version
Mono JIT compiler version 5.16.0.179 (tarball Thu Oct  4 10:29:10 UTC 2018)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           __thread
	SIGSEGV:       altstack
	Notifications: epoll
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug 
	Interpreter:   yes
	LLVM:          yes(3.6.0svn-mono-/)
	GC:            sgen (concurrent by default)
```

**Comment by Ronaldho80** [comment]
Duplicate of #9956
